src/main/java/org/broadinstitute/consent/http/service/DacService.java

```diff
@@ -294,7 +295,9 @@ List<DataAccessRequest> filterDataAccessRequestsByDac(List<DataAccessRequest> do
   }
 
   public Set<Dac> findByDatasetId(List<Integer> datasetIds) {
-    return dacDAO.findDacsForDatasetIds(datasetIds);
+    Set<Dac> returnSet = new HashSet<>();
+    dacDAO.findDacsForDatasetIds(datasetIds).forEach(dac -> returnSet.add(findById(dac.getDacId())));
```

Comment:
I generally try to avoid nested dao calls like this in favor of doing the necessary population in the outer dao call.